src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIMarsManager.cs

```diff
@@ -12,14 +11,14 @@ namespace System.Data.SqlClient.SNI
     /// </summary>
     internal class SNIMarsManager
     {
-        public static readonly SNIMarsManager Singleton = new SNIMarsManager();
-        private ConcurrentDictionary<SNIHandle, SNIMarsConnection> _connections = new ConcurrentDictionary<SNIHandle, SNIMarsConnection>();
+        private readonly ConcurrentDictionary<SNIHandle, SNIMarsConnection> _connections;
 
         /// <summary>
         /// Constructor
         /// </summary>
         public SNIMarsManager()
```

Comment:
I agree that it needs to be instanced. 
But do we need any mechanism to force the instance to be created only one time?
For example,
```
internal class SNIMarsManager
{
    ...
    private static bool instanced = false;
    public SNIMarsManager()
    {
        if (instanced)
        {
             // throw exception
        }
        // do constructor stuff
        instanced = true;
    }
    ...
}
```